src/storage/wal/catalog_delta_entry.cpp

```diff
@@ -484,4 +488,33 @@ void CatalogDeltaEntry::SaveState(TransactionID txn_id, TxnTimeStamp commit_ts)
     }
 }
 
+String CatalogDeltaEntry::ToString() const {
+    String str;
+    for (const auto &operation : operations_) {
+        str += operation->ToString() + "\n";
+    }
+    return str;
+}
+
+void GlobalCatalogDeltaEntry::Merge(UniquePtr<CatalogDeltaEntry> other) {
+
+    auto &global_map = this->encode_op_to_id_map_;
+    auto &global_operations = this->global_operations_;
+    auto &local_operations = other->operations();
+
+    for (auto &local_operation : local_operations) {
+        auto it = global_map.find(local_operation->EncodeIndex());
+        if (it == global_map.end()) {
+            global_map[local_operation->EncodeIndex()] = global_operations.size();
+            global_operations.push_back(std::move(local_operation));
+        } else {
+            global_operations[it->second] = std::move(local_operation);
```

Comment:
It looks too simple to merge local operation to the global operations. I am not sure if it is OK or not, that using new local operation to replace old operation in global operations.